docs/api/en/core/Object3D.html

```diff
@@ -413,15 +413,15 @@ <h3>[method:null traverse]( [param:Function callback] )</h3>
 		<p>
 		callback - A function with as first argument an object3D object.<br /><br />
 
-		Executes the callback on this object and all descendants.
+		Executes the callback on this object and all descendants. <b>Warning:</b> Do not add, move, or remove the object or its descendants while traversing.
 		</p>
 
 		<h3>[method:null traverseVisible]( [param:Function callback] )</h3>
 		<p>
 		callback - A function with as first argument an object3D object.<br /><br />
 
 		Like traverse, but the callback will only be executed for visible objects.
-		Descendants of invisible objects are not traversed.
+		Descendants of invisible objects are not traversed. <b>Warning:</b> Do not add, move, or remove the object or its descendants while traversing.
 		</p>
 
 		<h3>[method:null traverseAncestors]( [param:Function callback] )</h3>
```

Comment:
Should the description for `traverseAncestors()` be changed, too?

Reply:
Hm - do you think it's possible for it to break going upwards, too?

There's a check for `null` in [traverseAncestors](https://github.com/mjurczyk/three.js/blob/css2drenderer-outline/src/core/Object3D.js#L534). The only scenarios I can imagine rn is for the object to be either (1) moved to another parent or (2) removed from its parent. Case (1) should work fine and proceed, case (2) will just stop the traversal. 🤔

Reply:
I think it's more safe to just state that users should not change the object hierarchy while using `traverse*()` methods.